src/main/scala/nl/knaw/dans/api/sword2/DepositHandler.scala

```diff
@@ -152,51 +154,117 @@ object DepositHandler {
-    getFetchTxt(bagitDir).foreach(fetchText =>
-      { checkFetchTxtUrls(fetchText)
-        resolveFetchTxtFiles(bagitDir, fetchText)
-      }
-    )
-  }
-
   private def getFetchTxt(bagitDir: File)(implicit id: String): Option[FetchTxt] = Option {
     getBagFromDir(bagitDir).getFetchTxt
   }
 
-  private def checkFetchTxtUrls(fetchText: FetchTxt)(implicit id: String): Unit = {
+  def checkFetchTxtUrls(bagitDir: File)(implicit id: String): Try[Unit] = Try {
     log.debug(s"[$id] Checking validity of urls in fetch.txt")
-    fetchText.asScala.foreach(item =>
-      checkUrlValidity(item.getUrl))
+    getFetchTxt(bagitDir).foreach(fetchText =>
+      fetchText.asScala.foreach(item =>
+      checkUrlValidity(item.getUrl)))
   }
 
   private def checkUrlValidity(url: String)(implicit id: String): Unit = {
     // check if the url is syntactically correct
     try {
       new URL(url)
     } catch {
-      case e: MalformedURLException => throw new InvalidDepositException(id, s"Invalid url in Fetch Items ($url)")
+      case e: MalformedURLException => throw InvalidDepositException(id, s"Invalid url in Fetch Items ($url)")
     }
     // check if the url complies with the allowed url-structure
     val urlPattern = Pattern.compile(SwordProps("url-pattern"))
     if (!urlPattern.matcher(url).matches)
-      throw new InvalidDepositException(id, s"Not allowed url in Fetch Items ($url)")
+      throw InvalidDepositException(id, s"Not allowed url in Fetch Items ($url)")
   }
 
-  private def resolveFetchTxtFiles(bagitDir: File, fetchText: FetchTxt)(implicit id: String): Unit = {
-    log.debug(s"[$id] Resolving files in fetch.txt")
-    fetchText.asScala.foreach(item =>
-      Using.urlInputStream(new URL(item.getUrl)).foreach(src =>
-        Files.copy(src, Paths.get(bagitDir.getAbsolutePath, item.getFilename))))
-  }
-
-  def checkBagValidity(bagitDir: File)(implicit id: String): Try[Unit] = {
+  def checkBagValidity(bagitDir: File)(implicit id: String) = {
     log.debug(s"[$id] Verifying bag validity")
+
+    val fetchTexts = getFetchTxt(bagitDir).map(fetchText => fetchText.asScala).getOrElse(List()).toList
+    val fetchTextsBagStore = fetchTexts.filter(_.getUrl.startsWith(baseUrl))
+    val fetchTextsOther = fetchTexts diff fetchTextsBagStore
+
+    // resolve those FetchTxt files that do not refer to the bag store
+    resolveFetchTxtFiles(bagitDir, fetchTextsOther)
+
     val bag = getBagFromDir(bagitDir)
     val validationResult: SimpleResult = bag.verifyValid
-    if (validationResult.isSuccess) Success(Unit)
-    else Failure(new InvalidDepositException(id, validationResult.messagesToString))
+
+    if (fetchTextsBagStore.isEmpty) {
+      if (validationResult.isSuccess) Success(Unit)
+      else Failure(InvalidDepositException(id, validationResult.messagesToString))
+    }
+    else {
+      if (validationResult.isSuccess) log.warn(s"[$id] There is a fetch.txt file, while all the files are present in the bag.")
```

Comment:
Good point to note. Actually, isn't this an error? Is this something that could feasibly happen? If not, then it is *really* a serious error if it does, because it means the logic of our code turns out to be unsound.